src/tools/opt-dist/src/exec.rs

```diff
@@ -109,6 +109,8 @@ impl Bootstrap {
             "--stage",
             "2",
             "library/std",
+            // FIXME: benchmarks fails when using beta std on stage 0
+            "--build-std-on-stage0",
```

Comment:
Seems like opt-dist depends on stage 0 (old stage 0, I mean the compiled one) std atm.

cc @Kobzol

Reply:
`opt-dist` does the weird thing of running the test suite with an externally provided toolchain. This was "emulated" before by running stage 0 tests and providing an external rustc and cargo binaries.

What I want to do is build a (stage2, full) toolchain with bootstrap, and then run the test suite against that toolchain. But I can't do it in bootstrap directly, since the toolchain is optimized in the meantime, so I want to first build it, and then run tests that use the externally provided toolchain.